browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -50,12 +55,91 @@ bool ApplyPotentialReferrerBlock(std::shared_ptr<BraveRequestInfo> ctx) {
   return false;
 }
 
+bool ApplyPotentialQueryStringFilter(std::shared_ptr<BraveRequestInfo> ctx) {
+  DCHECK_CURRENTLY_ON(BrowserThread::IO);
+  if (!ctx->request_url.has_query()) {
+    return false;
+  }
+
+  std::string new_query = ctx->request_url.query();
+
+  bool modified = false;
+  for (auto tracker : query_string_trackers) {
+    size_t key_size, key_start;
+
+    // Look for the tracker anywhere in the query string
```

Comment:
It has been mentioned before that we should not be writing custom code for things like parsing query strings. If you're doing something related to urls, there is most likely a well tested method that already does what you want. In this case `net::QueryIterator` - see RewardsServiceImpl::OnXHRLoad for an example

Reply:
Agreed, but see my comments in https://github.com/brave/brave-core/pull/3239#issuecomment-524073918.

Reply:
The iterator you point to is based on the Mozilla parser I looked into: https://cs.chromium.org/chromium/src/net/base/url_util.cc?l=161&rcl=3fe839167bd0728a9715cba7aa23012e8b0efe3e